File: do-it-algorithm-coding-test-with-python/백준/Silver/1931. 회의실 배정/회의실 배정.py
import sys

input = sys.stdin.readline

N = int(input())

A = [[0] * 2 for _ in range(N)]

for i in range(N):
    A[i][1], A[i][0] = map(int, input().split())

A.sort()
count = 0
end = -1

for i in range(N):
    if A[i][1] >= end:
        end = A[i][0]
        count += 1

print(count)

File: 백준/Silver/1931. 회의실 배정/회의실 배정.js
const fs = require('fs');
const input = fs.readFileSync('./dev/stdin').toString().split('\n');

const n = Number(input[0]);

const times = [];

for (let i = 1; i <= n; i++) {
  times.push(input[i].split(' ').map(Number));
}

times.sort((a, b) => {
  if (a[1] === b[1]) return a[0] - b[0];
  return a[1] - b[1];
});

let end = 0;
let count = 0;

times.forEach(([s, e]) => {
  if (s >= end) {
    count++;
    end = e;
  }
});

console.log(count);
